heaps/min_heap.py

```diff
@@ -1,76 +1,106 @@
 class HeapNode:
-  
+
     def __init__(self, key, value):
         self.key = key
         self.value = value
 
     def __str__(self):
-        return str(self.value)
+        if self.value:
+            return str(self.value)
+        else:
+            return str(self.key)
 
     def __repr__(self):
         return str(self.value)
 
+
 class MinHeap:
 
     def __init__(self):
         self.store = []
+        self.length = 0
 
-
-    def add(self, key, value = None):
+    def add(self, key, value=None):
```

Comment:
👀 Make sure you are accounting for the case where Value is None

Space complexity would just be O(log n) here because of `heap_up`'s recursive call stack